### PR TITLE
Explicit Types for `->` iterators, fixes #205

### LINKED ISSUE #205: def => explicit types
Goal is to replace all uses of `def` with explicit types:

    // Before
    def count = 4
    def method() {
        return "string"
    }

    // After
    int count = 4
    String method() {
        return "string"
    }

The Groovy style guide indicates that it's a choice of using `def` or explicit types:

    So make a choice, either use def or a type.

Though `def` is more of the idiom for Groovy, explicit types help with correctness, catching errors in the IDE as a def isn't as constrained:

    When using def in Groovy, the actual type holder is Object (so you can assign any
    object to variables defined with def, and return any kind of object if a method is
    declared returning def).

Thus, a def is less restrictive on types. According to the following link in the Style Guide, it can be thought of as an alias to `Object` and the programmer indicating that they don't care about the type: http://www.groovy-lang.org/semantics.html#_variable_definition

### PR DESCRIPTION
**Pull request by brunobowden**
Depends on PR #220
- Add explicit types for `->` style iteration, fixes #205
- prefixProperties => packagePrefixes
- srcFiles => getSrcFiles() to make method call explicit
- Other similar cases as above
- ArrayList for {objcCompiler|linker}.withArguments iterators
- Configure: args => List<String> args

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - PTAL

This fixes up I think the last remaining issues to fix #205. For the `J2objcNativeCompilation` class, it wasn't clear what where the types of the objcCompiler and linker members. By using `pringln "args type: ${args.getClass()}", I was able to determine that they were both ArrayList. Though I wasn't able to confirm that independently.

This is the repeated section of code you should pay most attention to during the review:

    target('ios_arm64') {
        def iosClangArgs = [
                '-arch',
                'arm64']
        iosClangArgs += iphoneClangArgs
        objcCompiler.withArguments { ArrayList args ->
            iosClangArgs.each { args << it }
        }
        linker.withArguments { ArrayList args ->
            iosClangArgs.each { args << it }
        }
    }

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy [hunk 1]
```diff
@@ -99,10 +100,10 @@ class J2objcNativeCompilation {
                                     '-arch',
                                     'arm64']
                             iosClangArgs += iphoneClangArgs
-                            objcCompiler.withArguments { args ->
+                            objcCompiler.withArguments { ArrayList args ->
```
Comment (confile) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy]: Why not List?

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy [hunk 2]
```diff
@@ -99,10 +100,10 @@ class J2objcNativeCompilation {
...
                                 iosClangArgs.each { args << it }
                             }
-                            linker.withArguments { args ->
+                            linker.withArguments { ArrayList args ->
```
Comment (confile) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy]: List?

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy [hunk 3]
```diff
@@ -111,10 +112,10 @@ class J2objcNativeCompilation {
                                     '-arch',
                                     'armv7']
                             iosClangArgs += iphoneClangArgs
-                            objcCompiler.withArguments { args ->
+                            objcCompiler.withArguments { ArrayList args ->
```
Comment (confile) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy]: List?

**Comment by confile** [comment]
LGTM beside my comments

**Comment by advayDev1** [comment]
Per https://docs.gradle.org/current/javadoc/org/gradle/nativeplatform/toolchain/CommandLineToolConfiguration.html#withArguments(org.gradle.api.Action)

The args should be List<String>.  ArrayList works but doesn't specify the element type but is also too specific for the container type.

**Comment by advayDev1** [comment]
(btw, this branch is in upstream not your origin)

**Comment by brunobowden** [comment]
Rebase and now merging...

**Pull request merged**